### PR TITLE
Summary config from eclipsestate

### PR DESCRIPTION
**Pull request by jokva**

### CODE REVIEW

File: opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp [hunk 1]
```diff
@@ -449,6 +449,17 @@ namespace Opm {
         }
     }
 
+    std::vector< int > Eclipse3DProperties::getRegions( const std::string& kw ) const {
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp]: Can this be renamed to e.g. `getRegionValues()` - to more clearly differentiate from the full region keyword itself.

Comment (jokva) [inline opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp]: They're not *values* though, they're names/identifiers represented as an integer.

File: opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp [hunk 2]
```diff
@@ -189,27 +186,40 @@ namespace Opm {
...
+    SummaryConfig::SummaryConfig( const Deck& deck, const EclipseState& es )
+        : SummaryConfig( deck,
+                         *es.getSchedule(),
+                         es.get3DProperties(),
+                         dimensions( *es.getInputGrid() ) )
+    {}
+
+    SummaryConfig::SummaryConfig( const Deck& deck,
+                                  const Schedule& schedule,
+                                  const Eclipse3DProperties& props,
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.cpp]: We should augment the `Eclipse3DProperties` class with the `(nx,ny,nz)` information, that way we can shorten this signature slightly and it makes a certain sense to have there as well.

File: opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp (hunk 1 above)
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp]: > they're names/identifiers represented as an integer.

I agree - maybe `getRegionID` - or just leave it.

**Comment by joakim-hove** [comment]
Not really part of this patch - but in the `SummaryConfig` constructor there are several constructions of this type:

```C++
const auto regions = item.size() > 0 && item.hasValue( 0 )
            ? item.getData< int >()
            : es.getRegions( "FIPNUM" );
```
These keywords do not have a default specified, so the tests `item.size() > 0` and `itemt.hasValue( 0 )` must be both true - or alternatively both false.

**Comment by joakim-hove** [comment]
Extremely minor comments - fix if you agree; and I will merge.

**Comment by jokva** [comment]
> These keywords do not have a default specified, so the tests item.size() > 0 and itemt.hasValue( 0 ) must be both true - or alternatively both false.

Oh, ok. So by checking item.size() we can determine if we must look in the deck or not? I'll put it on the refactor list, that file will get some love soon.

**Pull request merged**